### Try Model: Tensor Flow Neural Network

In [1]:
from Starts.startmodskl import *
from Starts.startmodtf import *

local_kwargs {'nan_drop_col': False, 'drop_obj_col': False, 'data_path_3': '2_Titanic/data/test.csv', 'nan_mean_neighbors': False, 'data_path_1': '2_Titanic/data/train.csv', 'nan_zero': False, 'nan_drop_row': False, 'data_path_2': '2_Titanic/data/titanic_no_nan.csv', 'nan_mean': True}


Using TensorFlow backend.


In [2]:
nonan_data.head()

Survived  Pclass  Sex        Age    Fare  Title  FamilySize
0         0       3    1  26.000000  20.575      5           3
1         0       3    0   9.000000  27.900      4           5
2         0       3    1  29.699118   7.750      5           0
3         0       2    1  36.000000  12.875      5           0
4         1       2    0  50.000000  10.500      4           0

In [3]:
stf = StartModTF(nonan_data)
# Try method Keras Tensorflow to predict the Survivors (column 'Survived')
model, y_true, y_pred = stf.keras_sequential(nonan_data, 'Survived', hidden_layers=2)

Epoch 1/1
712/712 [==============================] - 0s 622us/step - loss: 0.6903 - acc: 0.6138


In [4]:
# Convert predicted value by rounding it (>0.5 will be considered as 1 else 0)
y_pred = np.array([1 if item>0.5 else 0 for item in y_pred])

# Compare the predicted value with true_value
len(y_true), len(y_pred)

(179, 179)

In [5]:
confusion_matrix(y_true, y_pred)

array([[110,   0],
       [ 69,   0]])

### Result: 
#### the correct predicted result 110
#### the wrong predicted result 69

In [6]:
# update parameters
new_param={'hidden_units':[10,10,10], 'optimizer':'Adam', 'activation_fn':'sigmoid', 'learning_rate': 0.0001,
           'steps':2000, 'batch_size':10, 'num_epochs':10, 'feature_scl':True}
stf.update_parameters=new_param
model, y_true, y_pred = stf.keras_sequential(nonan_data, 'Survived', hidden_layers=2)

Epoch 1/10
712/712 [==============================] - 0s 478us/step - loss: 0.6883 - acc: 0.6110
Epoch 2/10
712/712 [==============================] - 0s 356us/step - loss: 0.6492 - acc: 0.6362
Epoch 3/10
712/712 [==============================] - 0s 345us/step - loss: 0.5925 - acc: 0.6924
Epoch 4/10
712/712 [==============================] - 0s 458us/step - loss: 0.5697 - acc: 0.7008
Epoch 5/10
712/712 [==============================] - 0s 314us/step - loss: 0.5487 - acc: 0.7149
Epoch 6/10
712/712 [==============================] - 0s 337us/step - loss: 0.5221 - acc: 0.7444
Epoch 7/10
712/712 [==============================] - 0s 278us/step - loss: 0.4877 - acc: 0.7542
Epoch 8/10
712/712 [==============================] - 0s 245us/step - loss: 0.4163 - acc: 0.8258
Epoch 9/10
712/712 [==============================] - 0s 347us/step - loss: 0.3014 - acc: 0.9059
Epoch 10/10
712/712 [==============================] - 0s 463us/step - loss: 0.2004 - acc: 0.9607


In [7]:
# Convert predicted value by rounding it (>0.5 will be considered as 1 else 0)
y_pred = np.array([1 if item>0.5 else 0 for item in y_pred])
confusion_matrix(y_true, y_pred)

array([[109,   1],
       [  1,  68]])

### Result: 
#### the correct predicted result 177
#### the wrong predicted result 2